src/DataStructures/DataBox/PrefixHelpers.hpp

```diff
@@ -29,210 +29,100 @@ template <template <typename...> class Wrapper, typename TagList,
 using wrap_tags_in =
     tmpl::transform<TagList, tmpl::bind<Wrapper, tmpl::_1, tmpl::pin<Args>...>>;
 
-namespace DataBox_detail {
-enum class DispatchTagType {
-  Variables,
-  Prefix,
-  Other,
+namespace detail {
+template <template <typename...> class Prefix, typename VariablesTag,
+          typename... Args>
+struct variables_tag_with_tags_prefixed_by_impl {
+  static_assert(std::is_same_v<VariablesTag, void>,
+                "Template parameter VariablesTag is not a ::Tags::Variables");
 };
 
-template <typename Tag>
-constexpr DispatchTagType tag_type =
-    tt::is_a_v<Tags::Variables, Tag>
-        ? DispatchTagType::Variables
-        : std::is_base_of_v<db::PrefixTag, Tag> ? DispatchTagType::Prefix
-                                                : DispatchTagType::Other;
+template <template <typename...> class Prefix, typename TagList,
+          typename... Args>
+struct variables_tag_with_tags_prefixed_by_impl<
+    Prefix, ::Tags::Variables<TagList>, Args...> {
+  using type = ::Tags::Variables<wrap_tags_in<Prefix, TagList, Args...>>;
+};
+}  // namespace detail
 
-template <DispatchTagType TagType>
-struct add_tag_prefix_impl;
+/// \ingroup DataBoxTagsGroup
+/// Create a new ::Tags::Variables from the tags of VariablesTag, wrapping each
+/// tag with Prefix, which may have additional template arguments Args.
+template <template <typename...> class Prefix, typename VariablesTag,
+          typename... Args>
+using variables_tag_with_tags_prefixed_by =
```

Comment:
Isn't this metafunction just `db::add_tag_prefix` with the restriction that it accepts nothing but a Variables tag? I think you can just delete it and we use `db::add_tag_prefix` everywhere.